riotfile.py

```diff
@@ -1321,15 +1380,25 @@ def select_pys(min_version=MIN_PYTHON_VERSION, max_version=MAX_PYTHON_VERSION):
         Venv(
             name="fastapi",
             command="pytest {cmdargs} tests/contrib/fastapi",
+            pkgs={
+                "httpx": latest,
+                "pytest-asyncio": latest,
+                "requests": latest,
+                "aiofiles": latest,
+            },
             venvs=[
                 Venv(
-                    pys=select_pys(min_version="3.6"),
+                    pys=select_pys(min_version="3.6", max_version="3.10"),
                     pkgs={
                         "fastapi": [">=0.51,<0.52", ">=0.55,<0.56", ">=0.60,<0.61", latest],
-                        "httpx": latest,
-                        "pytest-asyncio": latest,
-                        "requests": latest,
-                        "aiofiles": latest,
+                    },
+                ),
+                Venv(
+                    # Python 3.11 only compatible with Starlette >=0.21 onwards.
+                    pys=select_pys(min_version="3.11"),
+                    pkgs={
+                        "fastapi": [latest],
+                        "starlette": latest,
```

Comment:
Can we add a comment why we impose a starlette version here? (It overrides the dependency installed by fastapi)